From 419c3b5bdfb963fb7e172cb899865662feab3102 Mon Sep 17 00:00:00 2001
Message-Id: <419c3b5bdfb963fb7e172cb899865662feab3102.1415496071.git.chang-joon.lee@intel.com>
In-Reply-To: <aa0c58fe046f9d00f3d381a75f6169b3f6d58c94.1415496071.git.chang-joon.lee@intel.com>
References: <aa0c58fe046f9d00f3d381a75f6169b3f6d58c94.1415496071.git.chang-joon.lee@intel.com>
From: Brad Volkin <bradley.d.volkin@intel.com>
Date: Wed, 21 May 2014 12:50:42 -0700
Subject: [PATCH 3/8] MUST_REBASE [VPG]: drm/i915: Drop oacontrol register
 tracking in cmd parser

The intention of the upstream change is to make sure every batch buffer
that enables OA performance monitoring also disables it. Discussion
indicated that the actual benefit was limited, but as upstream userspace
already did this and the patch was simple, it was merged.

This is MUST_REBASE because it would likely be more difficult than it is
worth to retrofit Android userspace to have this behavior.

For: APDEV-870
Change-Id: I93b6138c403fe57e2fe87c3478791ae183fed653
Signed-off-by: Brad Volkin <bradley.d.volkin@intel.com>
---
 drivers/gpu/drm/i915/i915_cmd_parser.c |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_cmd_parser.c b/drivers/gpu/drm/i915/i915_cmd_parser.c
index 1b3cab3..f3311d4 100644
--- a/drivers/gpu/drm/i915/i915_cmd_parser.c
+++ b/drivers/gpu/drm/i915/i915_cmd_parser.c
@@ -1077,11 +1077,6 @@ int i915_parse_cmds(struct intel_engine_cs *ring,
 		cmd += length;
 	}
 
-	if (oacontrol_set) {
-		DRM_DEBUG_DRIVER("CMD: batch set OACONTROL but did not clear it\n");
-		ret = -EINVAL;
-	}
-
 	if (cmd >= batch_end) {
 		DRM_DEBUG_DRIVER("CMD: Got to the end of the buffer w/o a BBE cmd!\n");
 		ret = -EINVAL;
-- 
1.7.9.5

